app/src/main/kotlin/com/hippo/ehviewer/coil/LimitConcurrencyNetworkClient.kt

```diff
@@ -14,13 +13,13 @@ import moe.tarsin.coroutines.NamedSemaphore
 import moe.tarsin.coroutines.withPermit
 
 class LimitConcurrencyNetworkClient(val impl: NetworkClient) : NetworkClient {
-    val semaphores = NamedSemaphore<String>(permits = Settings.thumbConcurrency)
+    val semaphores = NamedSemaphore<String>(permits = 8)
     override suspend fun <T> executeRequest(req: NetworkRequest, f: suspend (NetworkResponse) -> T): T {
         val url = req.url
         return when {
-            // Ex thumb server has no h2 multiplexing support
+            // Ex thumb server may not have h2 multiplexing support
             URL_PREFIX_THUMB_EX in url -> semaphores.withPermit(URL_PREFIX_THUMB_EX) { withContext(NonCancellable) { impl.executeRequest(req, f) } }
-            // H@H server has no h2 multiplexing support
+            // H@H server have no h2 multiplexing support
```

Comment:
We don't know actually.
I tested 5 servers and they all support h2.
I guess maybe *.hath.network are new servers.